src/Dev/Tasks/MigrateFileTask.php

```diff
@@ -46,85 +46,102 @@ public function run($request)
         $args = $request->getVars();
         $this->validateArgs($args);
 
+        $this->extend('preFileMigration');
+
         $subtasks = !empty($args['only']) ? explode(',', $args['only']) : $this->defaultSubtasks;
 
-        if (in_array('move-files', $subtasks)) {
+        $subtask = 'move-files';
+        if (in_array($subtask, $subtasks)) {
             if (!class_exists(FileMigrationHelper::class)) {
                 $this->logger->error("No file migration helper detected");
-                return;
-            }
-
-            $this->logger->info('### Migrating filesystem and database records (move-files)');
-
-            $this->logger->info('If the task fails or times out, run it again and it will start where it left off.');
-
-            $migrated = FileMigrationHelper::singleton()->run();
-            if ($migrated) {
-                $this->logger->info("{$migrated} File DataObjects upgraded");
             } else {
-                $this->logger->info("No File DataObjects need upgrading");
+                $this->extend('preFileMigrationSubtask', $subtask);
+                $this->logger->info('### Migrating filesystem and database records (move-files)');
```

Comment:
Another opportunity for your new `$subtask` variable.

```suggestion
                $this->logger->info("### Migrating filesystem and database records ({$subtask})");
```